src/main/kotlin/no/nav/syfo/consumer/arbeidsgiverNotifikasjonProdusent/ArbeidsgiverNotifikasjonProdusent.kt

```diff
@@ -0,0 +1,114 @@
+package no.nav.syfo.consumer.arbeidsgiverNotifikasjonProdusent
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.UrlEnv
+import no.nav.syfo.auth.AzureAdTokenConsumer
+import no.nav.syfo.consumer.narmesteLeder.NarmesteLederConsumer
+import org.slf4j.LoggerFactory
+
+open class ArbeidsgiverNotifikasjonProdusent(urlEnv: UrlEnv, azureAdTokenConsumer: AzureAdTokenConsumer, narmesteLederConsumer: NarmesteLederConsumer) {
+    private val client: HttpClient
+    private val azureAdTokenConsumer: AzureAdTokenConsumer
+    private val narmesteLederConsumer: NarmesteLederConsumer
+    private val arbeidsgiverNotifikasjonProdusentBasepath: String
+    private val log = LoggerFactory.getLogger("no.nav.syfo.consumer.AgNotifikasjonProdusentConsumer")
+    private val scope = urlEnv.arbeidsgiverNotifikasjonProdusentApiScope
+    private val dineSykmeldteUrl = urlEnv.baseUrlDineSykmeldte
+    private val EMAIL_BODY = EMAIL_BODY_START + dineSykmeldteUrl + EMAIL_BODY_END
+
+    init {
+        client = HttpClient(CIO) {
+            expectSuccess = false
+            install(JsonFeature) {
+                serializer = JacksonSerializer {
+                    registerKotlinModule()
+                    registerModule(JavaTimeModule())
+                    configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                }
+            }
+        }
+        this.azureAdTokenConsumer = azureAdTokenConsumer
+        this.narmesteLederConsumer = narmesteLederConsumer
```

Comment:
Nå kan vel `azureAdTokenConsumer ` og `narmesteLederConsumer ` fjernes herfra?

Reply:
Glemte å fjerne `narmesteLederConsumer ` 😬 , men `azureAdTokenConsumer` er i bruk for å hente bearer token i `callArbeidsgiverNotifikasjonProdusent(...)`